forge/forge/components/file_manager/file_manager.py

```diff
@@ -13,7 +16,21 @@
 logger = logging.getLogger(__name__)
 
 
-class FileManagerComponent(DirectiveProvider, CommandProvider):
+class FileManagerConfiguration(BaseModel):
+    files_path: str
+    """Path to agent files, e.g. state"""
+    workspace_path: str
+    """Path to files that agent has access to"""
+
+    class Config:
+        # Prevent mutation of the configuration
+        # as this wouldn't be reflected in the file storage
+        allow_mutation = False
+
+
+class FileManagerComponent(
```

Comment:
Should the file manager be responsable for storing state or should a state manager component require file manager?

Reply:
I don't think so, splitting file manager to state manager is on my todo list.